src/protocol_muxer/multiselect/multiselect_instance.cpp

```diff
@@ -166,6 +181,10 @@ namespace libp2p::protocol_muxer::multiselect {
 
   void MultiselectInstance::close(outcome::result<std::string> result) {
     closed_ = true;
+
+    // Cancel timeout if it's still active
+    timeout_handle_.reset();
```

Comment:
Should we `reset()` connection/stream, to cancel pending read/write?

Reply:
I think it will be automatically reset, once we erase this active instance:

https://github.com/libp2p/cpp-libp2p/blob/4bd3a7826dbfdc1350015345351bb82b5c01917e/src/protocol_muxer/multiselect.cpp#L56-L64

It might live a little longer in the cache, but eventually a new connection will replace the one from the instance